cpp/src/arrow/flight/transport_impl.h

```diff
@@ -0,0 +1,310 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+/// \file
+/// Internal (but not private) interface for implementing
+/// alternate network transports in Flight.
+///
+/// \warning EXPERIMENTAL. Subject to change.
+///
+/// To implement a transport, implement ServerTransportImpl and
+/// ClientTransportImpl, and register the desired URI schemes with
+/// TransportImplRegistry. Flight takes care of most of the per-RPC
+/// details; transports only handle connections and providing a I/O
+/// stream implementation (TransportDataStream).
+///
+/// On the server side:
+///
+/// 1. Applications subclass FlightServerBase and override RPC handlers.
+/// 2. FlightServerBase::Init will look up and create a ServerTransportImpl
+///    based on the scheme of the Location given to it.
+/// 3. The ServerTransportImpl will start the actual server. (For instance,
```

Comment:
Naming-wise, why do we have `FlightServerBase` and `ServerTransportImpl`? I would expect a consistent terminology, for example always "Base" or always "Impl" (in both cases, these are a base abstract class that the user needs to implement by deriving it).

Reply:
No particular reason, I can rename to ServerTransportBase or frankly just ServerTransport.